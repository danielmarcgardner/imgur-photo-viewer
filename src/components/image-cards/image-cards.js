import React from 'react';
import { withRouter } from 'react-router-dom';
import '../../App.css';

const ImageCard = ({ image, id, history }) => (
  <div className="IndividualCard" onClick={() => history.push(`/hero/${id}`)}>
    <div className="ImageHeader">
      <img src={image} alt={id} className="Image" />
    </div>
  </div>
);

export default withRouter(ImageCard);
