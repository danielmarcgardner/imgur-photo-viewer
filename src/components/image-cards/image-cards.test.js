import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ImageCard } from './image-cards';

configure({ adapter: new Adapter() });

const image = 'https://i.imgur.com/fZ8JNw4.jpg';
const id = 'fZ8JNw4';
const history = {
  push: jest.fn(),
};

describe('ImageCard', () => {
  it('Renders to page', () => {
    const component = shallow(
      <ImageCard
        image={image}
        id={id}
        history={history}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('When the ImageCard is clicked, history.push is called and with the proper url', () => {
    const component = mount(
      <ImageCard
        image={image}
        id={id}
        history={history}
      />,
    );
    component.find('.IndividualCard').simulate('click');
    expect(history.push).toHaveBeenCalled();
    expect(history.push.mock.calls[0][0]).toBe('/hero/fZ8JNw4');
  });
  it('Renders with all correct properties', () => {
    const component = mount(
      <ImageCard
        image={image}
        id={id}
        history={history}
      />,
    );

    expect(component.find('.IndividualCard').length).toBe(1);
    expect(component.find('.ImageHeader').length).toBe(1);
    expect(component.find('.Image').length).toBe(1);
  });
});
