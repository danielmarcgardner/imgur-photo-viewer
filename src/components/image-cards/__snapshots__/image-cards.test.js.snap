// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard Renders to page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    history={
        Object {
            "push": [Function],
          }
    }
    id="fZ8JNw4"
    image="https://i.imgur.com/fZ8JNw4.jpg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ImageHeader"
>
        <img
                alt="fZ8JNw4"
                className="Image"
                src="https://i.imgur.com/fZ8JNw4.jpg"
        />
</div>,
      "className": "IndividualCard",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="fZ8JNw4"
          className="Image"
          src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
        "className": "ImageHeader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "fZ8JNw4",
          "className": "Image",
          "src": "https://i.imgur.com/fZ8JNw4.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ImageHeader"
>
          <img
                    alt="fZ8JNw4"
                    className="Image"
                    src="https://i.imgur.com/fZ8JNw4.jpg"
          />
</div>,
        "className": "IndividualCard",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="fZ8JNw4"
            className="Image"
            src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
          "className": "ImageHeader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "fZ8JNw4",
            "className": "Image",
            "src": "https://i.imgur.com/fZ8JNw4.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
